draft-ietf-ipsecme-ikev2-multiple-ke-00.xml

```diff
@@ -1043,7 +1055,7 @@ Type     Description                   Used In
         <t> Responders may consider recommendations from <xref target="RFC8019" />
         to deal with increased DoS attack susceptibility.  It is also possible that 
         the responder only agrees to create initial IKE SA without performing 
-        additional key exchanges, provided the initiator includes such an option in its proposals.  Then peers immediately rekey 
+        additional key exchanges, provided the initiator includes such an option in its proposals, i.e. support for childless IKE SA <xref target="RFC6023"/>.  Then peers immediately rekey 
```

Comment:
That's not correct. According to the RFC 6023 it is the responder which indicates the support for this option (by including the CHILDLESS_IKEV2_SUPPORTED notify into the response) and the ultimate decision whether to create Child SA or not is taken by the initiator. I think this piece of text needs some rewording. Below is not the new text, but just an outline of the peers' behaviour.

First, the initiator must include two proposals into the SA payload - one with Additional Key Exchange transforms and the other - without them (it is also possible that the initiator includes only one proposal without Additional Key Exchange transforms, but in this case the responder doesn't know whether the initiator ever supports multiple key exchanges, so it might be inappropriate for its local policy). If the responder doesn't want to perform multiple key exchanges at the time IKE SA is being created, it will select the proposal containing no Additional Key Exchange transforms and the IKE SA will be created as usual. The responder may also indicate that Childless IKE SA is supported, in which case the initiator may omit creating Child SA in IKE_AUTH, but actually the responder cannot force the initiator not to try to create the Child SA. If the initiator still tries to create Child SA, the responder may send some non-fatal error notify like NO_PROPOSAL_CHOSEN in the IKE_AUTH response. After the IKE SA is created it should be immediately rekeyed with Additional Key Exchanges. After that the initiator requests creating Child SA via CREATE_CHILD_SA.

Reply:
Thanks Valery, I think the correct text should go into Section 3.2.5. I'll add this later.

For this particular sentence that you picked up, it is part of Section 5 (Security Considerations), perhaps we can just a simplified sentence, e.g.:

```suggestion
        additional key exchanges by indicating the use of childless IKE SA <xref target="RFC6023"/>.  Then peers immediately rekey 
```

Reply:
I'd rather to retain the original text. The key point is that the initiator should allow the possibility to create IKE SA without performing additional key exchanges at that time (by including a proposal without Additional Key Exchange transforms). Whether or not Child SA is created is not so important. It is possible that the Child SA is created in IKE_AUTH, but no traffic is sent over it until the IKE SA is rekeyed and then a new fully protected Child SA is created. Creating childless IKE SA in this case is a better solution, then the scenario described above, but not mandatory.

Reply:
Ok, that makes sense. I've reverted it to the original text. I was thinking of "hinting" on the use of RFC6023, but you are right, there is no guarantee that this RFC is widely supported (as it's an experimental one) and one can achieve the same by immediately rekeying the IKE and Child SAs.